R/sce_to_anndata.R

```diff
@@ -60,10 +60,35 @@ sce_to_anndata <- function(
   # assign SCE to new variable to avoid modifying input SCE
   sce_to_convert <- sce
 
-  # remove any objects or dataframes
+  # keep only vectors and data frames; no objects or DataFrames
   metadata_to_keep <- metadata(sce_to_convert) |>
-    purrr::discard(is.object) |>
-    purrr::discard(is.list)
+    purrr::keep(\(x) {
+      is.atomic(x) || is.data.frame(x)
+    })
+
+
+  # prepare data frames for anndata conversion:
+  # - remove any list columns
+  # - ensure any NA values in character columns are formatted as character
+  # - ensure any
+  metadata_to_keep <- metadata_to_keep |>
+    purrr::map(
+      \(meta) {
```

Comment:
Super minor, but I find calling the variable here `meta` to be confusing, because the whole thing is the metadata, and this is just an item within that list. Using `x` as you do above (consistency!) seems fine to me.
```suggestion
      \(x) {
```